stripped/colabs/Analiza_szeregow_czasowych_dot_roslinnosci_stripped.py

```diff
@@ -0,0 +1,180 @@
+"""
+
+"""
+Utworzenie środowiska pyspark do obliczeń:
+"""
+
+!apt-get install openjdk-8-jdk-headless -qq > /dev/null
+!wget -q dlcdn.apache.org/spark/spark-3.5.0/spark-3.5.0-bin-hadoop3.tgz
+!tar xf spark-3.5.0-bin-hadoop3.tgz
+!pip install -q findspark
+
+
+import os
+os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"
+os.environ["SPARK_HOME"] = "/content/spark-3.5.0-bin-hadoop3"
+import findspark
+findspark.init()
+from pyspark.sql import SparkSession
+from google.colab import drive
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType
+import pandas as pd
+
+
+"""
+Utowrzenie sesji:
+"""
+
+spark = SparkSession.builder\
+        .master("local")\
+        .appName("Colab")\
+        .config('spark.ui.port', '4050')\
+        .getOrCreate()
+
+
+"""
+Połączenie z dyskiem:
+"""
+
+drive.mount('/content/drive')
+
+
+"""
+Wczytanie danych NASA znajdujących się na dysku w sparku:
+"""
+
+# Wczytanie zbioru sampled w celu pobrania nazw kolumn
+sampled = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/sampled_NASA_200k.csv')
+
+# Utworzenie schematu określającego typ zmiennych
+schemat = StructType()
+for i in sampled.columns:
+  if i == "Date":
+    schemat = schemat.add(i, StringType(), True)
+  else:
+    schemat = schemat.add(i, FloatType(), True)
+
+
+# Wczytanie zbioru Nasa w sparku
+nasa = spark.read.format('csv').option("header", True).schema(schemat).load('/content/drive/MyDrive/BigMess/NASA/NASA.csv')
+
+
+"""
+Zanim zacznimy pisać kwerendy należy jeszcze dodać nasz DataFrame (df) do "przestrzeni nazw tabel" Sparka:
+"""
+
+nasa.createOrReplaceTempView("nasa")
+
+
+"""
+Rozdzielenie kolumny "Date" na kolumny "Year" oraz "Month"
+"""
+
+nasa = spark.sql("""
+          SELECT
+          CAST(SUBSTRING(CAST(Date AS STRING), 1, 4) AS INT) AS Year,
+          CAST(SUBSTRING(CAST(Date AS STRING), 5, 2) AS INT) AS Month,
+          n.*
+          FROM nasa n
+          """)
+
+
+nasa = nasa.drop("Date")
+nasa.createOrReplaceTempView("nasa")
+
+
+"""
+# Analiza szergów czasowych  dot. roślinności
+"""
+
+# Import biblotek
+import matplotlib.pyplot as plt
+import random
+import matplotlib.gridspec as gridspec
+
+
+# Ograniczenie zbioru nasa do wartości, które potrzebujemy, ustawienie pomiaru zawsze na 1 dzień miesiąca
+szeregNasa = spark.sql("""
+                      SELECT
+                      lon, lat,
+                      to_date(CONCAT(Year, '-', Month, '-1')) as Date, GVEG, LAI
+                      FROM nasa
+                      order by lon, lat, Date
+                      """)
+
+
+%%time
+szeregNasa.show(5)
+
+
+# Wyzanczenie unikatowych par współrzednych ze zbioru Nasa i zapisanie w Pandas
+%%time
+distinct_wsp = spark.sql("""
+                          SELECT DISTINCT lon, lat from nasa
+                          """).toPandas()
+
+
+distinct_wsp.shape
+
+
+# Funkcja generująca wykres w czasie wartości zmiennej GVEG - green vegetation dla zadanych współrzednych
+def wykres_szeregu(lon, lat):
```

Comment:
takie funkcje bedziemy sobie wyciagac do osobnego pliku. Poki nie pokazałem jak to zrobic dodajmy przed tą funkcją wpis
```
#FIXME przenieść do katalogu src / @see (link do zadania)
```
w miejsce link do zadania wklej link do zadania z backlogu z opisem aby wydzielic funkcje do libów (src) z projektu.